components/script/dom/mediafragmentparser.rs

```diff
@@ -114,8 +114,8 @@ impl MediaFragmentParser {
             }
         } else {
             let mut iterator = fragment.split(',');
-            let start = parse_hms(iterator.next().ok_or_else(|| ())?)?;
-            let end = parse_hms(iterator.next().ok_or_else(|| ())?)?;
+            let start = parse_hms(iterator.next().ok_or(()))?;
```

Comment:
I think the build error you are having originates here, since it expects an `&str` but it is receiving a `Result`. One way to solve it would be adding back the second `?`, as it will unwrap the `Result` type. You can probably also get rid of the `ok_or(())` operator since `?` should also work for `Option` types and we are not using the error type. This goes for lines 117, 118, 317, 318, 322 and 333.

Reply:
Yeah I fixed  all 6 of them  already…  I’m trying to build it before pushing to be sure there no more errors Thanks @eerii

Reply:
The PR is ready to go now

Reply:
A member from the servo team will have to approve the pr but all the changes are looking good to me, thanks!